### PR TITLE
add no-empty-nested-blocks rule

### LINKED ISSUE #145: Add rule "Nested blocks of code should not be left empty"
[RSPEC-108](https://jira.sonarsource.com/browse/RSPEC-108)

### PR DESCRIPTION
**Pull request by stas-vilchik**
fixes #145

### CODE REVIEW

File: docs/rules/no-empty-nested-blocks.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+##  Noncompliant Code Example
+
+```typescript
+for (let i = 0; i < 42; i++) {}  // Empty on purpose or missing piece of code ?
+```
+
+## Exceptions
+
+When a block contains a comment, this block is not considered to be empty.
```
Comment (vilchik-elena) [inline docs/rules/no-empty-nested-blocks.md]: @stas-vilchik + catch blocks

File: src/rules/noEmptyNestedBlocksRule.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      !this.isCatchClause(node.parent) &&
+      !this.hasLeadingComment(node.parent)
+    ) {
+      this.addFailureAtNode(node, Rule.MESSAGE);
+    }
+
+    super.visitBlock(node);
+  }
+
+  private isLikeFunction(node?: ts.Node): boolean {
```
Comment (vilchik-elena) [inline src/rules/noEmptyNestedBlocksRule.ts]: @stas-vilchik why such name? why not just `isFunctionBodyBlock`?

File: tslint-sonarts.json [hunk 3]
```diff
@@ -15,6 +15,7 @@
     "no-misspelled-operator": true,
     "no-inconsistent-return": false,
     "no-unconditional-jump": true,
-    "no-misleading-array-reverse": true
+    "no-misleading-array-reverse": true,
+    "no-empty-nested-blocks": true
```
Comment (vilchik-elena) [inline tslint-sonarts.json]: @stas-vilchik let's disable it, as similar rule is part of tslint recommended profile.

**Comment by vilchik-elena** [comment]
@stas-vilchik feel free to merge when my remarks are fixed

File: tslint-sonarts.json (hunk 3 above)
Comment (stas-vilchik) [inline tslint-sonarts.json]: @vilchik-elena but why do we implement this rule then?

File: src/rules/noEmptyNestedBlocksRule.ts (hunk 2 above)
Comment (stas-vilchik) [inline src/rules/noEmptyNestedBlocksRule.ts]: @vilchik-elena it is not checking the function body block, it's checking the parent of the function body block.

File: docs/rules/no-empty-nested-blocks.md (hunk 1 above)
Comment (stas-vilchik) [inline docs/rules/no-empty-nested-blocks.md]: @vilchik-elena +class methods +arrow functions +function declarations

do I add them all? do I change the RSPEC as well?

**Pull request merged**